### PR TITLE
Use separate metric file names for same tests running on different platforms

### LINKED ISSUE #101220: Samsung S10 results are clobbering(?) Moto G4 results in SkiaPerf
The worst_frame_rasterizer_time_millis result for Moto G4 for commit https://github.com/flutter/flutter/commit/28505569463d5510f0feadd53e3ed10726e9b227 in SkiaPerf is recorded as 18.094:

https://flutter-flutter-perf.skia.org/e/?begin=1648581332&end=1648855266&keys=X02af86de85ec7fab80fc22a11016955d&requestType=0

However examining the logs in LUCI, this is neither the result reported for the S10 nor the Moto G4:

S10: https://storage.googleapis.com/flutter_logs/flutter/28505569463d5510f0feadd53e3ed10726e9b227/backdrop_filter_perf__timeline_summary/5d57a33d-a31f-4235-b1f5-de6bc493be8c/backdrop_filter_perf.timeline_summary.json

Moto G4: https://storage.googleapis.com/flutter_logs/flutter/28505569463d5510f0feadd53e3ed10726e9b227/backdrop_filter_perf__timeline_summary/4b729b7c-8ba4-4f57-a616-f95e391439f2/backdrop_filter_perf.timeline_summary.json

But the value recorded in SkiaPerf is close to the S10 result, which makes me think that something is being clobbered, or some wires are crossed somewhere.

Looking back at the results of the benchmark, it looks like this started happening around the time of the commit here: https://github.com/flutter/flutter/commit/ff425fc733018d183a452330de8ffb85e0689862:

https://flutter-flutter-perf.skia.org/e/?begin=1644280799&end=1648855266&queries=device_type%3DMoto_G__4_%26sub_result%3D90th_percentile_frame_rasterizer_time_millis%26sub_result%3D99th_percentile_frame_rasterizer_time_millis%26sub_result%3Daverage_frame_rasterizer_time_millis%26sub_result%3Dworst_frame_rasterizer_time_millis%26test%3Dbackdrop_filter_perf__e2e_summary%26test%3Dbackdrop_filter_perf__timeline_summary&requestType=0

Prioritizing as P2 since we are basically flying blind on lower end devices if this mixup is happening everywhere that we're running benchmarks on S10 devices.

@keyonghan

### PR DESCRIPTION
**Pull request by keyonghan**
Same tests may run on different platforms, for example: `backdrop_filter_perf__timeline_summary` is running on both `motoG4` and `samsungS10`.

Existing logic shares same metric file for both platforms, and causes confusion. This PR uses separate metric files for different cases.

Potential fix: https://github.com/flutter/flutter/issues/101220

### CODE REVIEW

File: dev/devicelab/lib/framework/metrics_center.dart [hunk 1]
```diff
@@ -133,5 +133,43 @@ Future<void> uploadToSkiaPerf(String? resultsPath, String? commitTime, String? t
...
+/// real host/device tags.
+/// 
+/// For example:
+///   Old file name: `backdrop_filter_perf__timeline_summary`
+///   New file name: `backdrop_filter_perf__timeline_summary_intel_linux_motoG4`
+String metricFileName(
+  String taskName,
+  Map<String, dynamic> benchmarkTagsMap,
+) {
+  String fileName = taskName;
```
Comment (zanderso) [inline dev/devicelab/lib/framework/metrics_center.dart:159]: See `StringBuffer`: https://api.dart.dev/dev/2.17.0-270.0.dev/dart-core/StringBuffer-class.html

Comment (keyonghan) [reply dev/devicelab/lib/framework/metrics_center.dart]: Thanks for the reference! Used `StringBuffer` instead.

**Review by zanderso — APPROVED** [review]

**Comment by zanderso** [comment]
Something is going wrong with the Google check. It might need to be landed manually.

**Comment by keyonghan** [comment]
Landing manually

**Pull request merged**